### PR TITLE
refactor: use dbconfig package to simplify the config generation

### PR DESCRIPTION
**Pull request by zyy17**

### CODE REVIEW

File: controllers/greptimedbcluster/deployers/datanode.go [hunk 1]
```diff
@@ -218,21 +199,6 @@ func (d *DatanodeDeployer) generateSvc(cluster *v1alpha1.GreptimeDBCluster) (*co
 					Protocol: corev1.ProtocolTCP,
 					Port:     cluster.Spec.MySQLServicePort,
 				},
-				{
-					Name:     "postgres",
-					Protocol: corev1.ProtocolTCP,
-					Port:     cluster.Spec.PostgresServicePort,
```
Comment (daviderli614) [inline controllers/greptimedbcluster/deployers/datanode.go:224]: question: if deleted, unnecessary define in [type](https://github.com/GreptimeTeam/greptimedb-operator/blob/develop/apis/v1alpha1/greptimedbcluster_types.go#L445) ?

File: controllers/greptimedbcluster/deployers/datanode.go [hunk 2]
```diff
@@ -411,16 +354,6 @@ func (d *DatanodeDeployer) generatePodTemplateSpec(cluster *v1alpha1.GreptimeDBC
 			Protocol:      corev1.ProtocolTCP,
 			ContainerPort: cluster.Spec.PostgresServicePort,
 		},
```
Comment (daviderli614) [inline controllers/greptimedbcluster/deployers/datanode.go:356]: suggestion: The service not define the postgres port, pod template spec not need to be defined[here](https://github.com/GreptimeTeam/greptimedb-operator/blob/2167a0996ced882d80e2bbb8eb8130b888178014/controllers/greptimedbcluster/deployers/datanode.go#L353)

**Comment by daviderli614** [comment]
question: Users use ak/sk to access aws s3, there is no interface, only to modify the toml file to achieve.

File: cmd/initializer/main.go [hunk 3]
```diff
@@ -15,141 +15,32 @@
...
 
-	// TODO(zyy17): We should add the flag to distinguish different components, not just datanode.
-	var datanodeConfig DatanodeConfig
-	_, err := toml.DecodeFile(opts.defaultConfigPath, &datanodeConfig)
-	if err != nil {
-		klog.Fatalf("Parse input toml file failed, err '%v'", err)
-	}
-
-	if opts.storageType == "S3" {
-		datanodeConfig.StorageConfig.Type = "S3"
```
Comment (daviderli614) [inline cmd/initializer/main.go:83]: question: Users use ak/sk to access aws s3, there is no interface, only to modify the toml file to achieve.

File: controllers/greptimedbcluster/deployers/datanode.go (hunk 1 above)
Comment (zyy17) [reply controllers/greptimedbcluster/deployers/datanode.go:224]: The datanode service doesn't need pg or mysql port anymore. It just need HTTP and gRPC service. However, the frontend service still need those settings.

File: cmd/initializer/main.go (hunk 3 above)
Comment (zyy17) [reply cmd/initializer/main.go:83]: User still can use secret to setup their S3 credentials.

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**